From ef560a735f50f8f2b3b5a7b7dc6cb1d361e11cdf Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Mon, 19 Mar 2018 12:32:01 -0700
Subject: [PATCH] OS-6742 "vmadm reboot" is not implemented for bhyve Reviewed
 by: Dave Eddy <dave.eddy@joyent.com> Reviewed by: Trent Mick
 <trentm@gmail.com> Approved by: Trent Mick <trentm@gmail.com>

---
 src/vm/node_modules/VM.js | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 6c0f3463..fb2bbb7d 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -14429,21 +14429,21 @@ function reset(uuid, log, callback)
 }
 
 /*
- * This handles rebooting docker=true zones.
+ * This handles rebooting docker=true and bhyve zones.
  */
-function doDockerReboot(vmobj, options, callback)
+function doStopStartReboot(vmobj, options, callback)
 {
     var log = options.log;
     var tracers_obj;
 
     if (process.env.EXPERIMENTAL_VMJS_TRACING) {
-        tracers_obj = traceUntilCallback('docker-reboot', log, callback);
+        tracers_obj = traceUntilCallback('stopstart-reboot', log, callback);
         callback = tracers_obj.callback;
         log = tracers_obj.log;
     }
 
     log.debug({vmobj_pid: vmobj.pid, timeout: options.timeout},
-        'doDockerReboot');
+        'doStopStartReboot');
 
     if (!options.timeout) {
         options.timeout = 10;
@@ -14680,8 +14680,8 @@ exports.reboot = function (uuid, options, callback)
             // re-add log in case it was changed
             options.log = log;
 
-            if (vmobj.docker) {
-                doDockerReboot(vmobj, options, cb);
+            if (vmobj.docker || vmobj.brand === 'bhyve') {
+                doStopStartReboot(vmobj, options, cb);
             } else {
                 doReboot(vmobj, options, cb);
             }
-- 
2.21.0

